Querying for non-existent record by constrained field panics

**Describe the bug**
Querying for non-existent record panics in worker

**To Reproduce** 
```golang
type Config struct {
	Gid  uint64 `json:"gid,omitempty"`
	Name string `json:"name,omitempty"`
}

func main() {
	engine, err := modusdb.NewEngine(modusdb.NewDefaultConfig("./db"))
	if err != nil {
		log.Fatalf("Failed to create engine: %v", err)
	}
	defer engine.Close()

	ctx := context.Background()

	gid, config, err := modusdb.Get[Config](ctx, engine,
		modusdb.ConstrainedField{
			Key:   "name",
			Value: "main",
		},
	)
```

```
panic: runtime error: integer divide by zero

goroutine 97 [running]:
github.com/hypermodeinc/dgraph/v24/worker.(*groupi).connToZeroLeader(0x1064dc1e0)
        /Users/matthew/go/pkg/mod/github.com/hypermodeinc/dgraph/v24@v24.0.3-0.20250123224129-a0d027dcffe0/worker/groups.go:738 +0x374
github.com/hypermodeinc/dgraph/v24/worker.(*groupi).BelongsToReadOnly(0x1064dc1e0, {0x1400053c120, 0xd}, 0x2)
        /Users/matthew/go/pkg/mod/github.com/hypermodeinc/dgraph/v24@v24.0.3-0.20250123224129-a0d027dcffe0/worker/groups.go:440 +0x164
github.com/hypermodeinc/dgraph/v24/worker.ProcessTaskOverNetwork({0x1058a1718, 0x1400031e330}, 0x14000494000)
        /Users/matthew/go/pkg/mod/github.com/hypermodeinc/dgraph/v24@v24.0.3-0.20250123224129-a0d027dcffe0/worker/task.go:137 +0x54
github.com/hypermodeinc/dgraph/v24/query.ProcessGraph({0x1058a1718, 0x1400031e330}, 0x1400051a908, 0x0, 0x1400069caf0)
        /Users/matthew/go/pkg/mod/github.com/hypermodeinc/dgraph/v24@v24.0.3-0.20250123224129-a0d027dcffe0/query/query.go:2170 +0x870
created by github.com/hypermodeinc/dgraph/v24/query.(*Request).ProcessQuery in goroutine 1
        /Users/matthew/go/pkg/mod/github.com/hypermodeinc/dgraph/v24@v24.0.3-0.20250123224129-a0d027dcffe0/query/query.go:2884 +0xaa8
```

**Expected behavior** 
Should be able to query for a non-existent type and have it return a not-found error

**Additional Context**
I believe this is because there's no schema applied before this query. And the worker is trying to find the group the predicates belong to.

**Environment**

- OS: macOS
- Language Go

--- a/.trunk/trunk.yaml
+++ b/.trunk/trunk.yaml
@@ -16,7 +16,7 @@ plugins:
 # Many linters and tools depend on runtimes - configure them here. (https://docs.trunk.io/runtimes)
 runtimes:
   enabled:
-    - go@1.23.3
+    - go@1.24.0
     - node@18.20.5
     - python@3.10.8
 

--- a/api.go
+++ b/api.go
@@ -7,7 +7,8 @@ package modusdb
 
 import (
 	"context"
-	"fmt"
+	"errors"
+	"reflect"
 
 	"github.com/hypermodeinc/dgraph/v24/dql"
 	"github.com/hypermodeinc/dgraph/v24/schema"
@@ -20,7 +21,7 @@ func Create[T any](ctx context.Context, engine *Engine, object T,
 	engine.mutex.Lock()
 	defer engine.mutex.Unlock()
 	if len(nsId) > 1 {
-		return 0, object, fmt.Errorf("only one namespace is allowed")
+		return 0, object, errors.New("only one namespace is allowed")
 	}
 	ctx, ns, err := getDefaultNamespace(ctx, engine, nsId...)
 	if err != nil {
@@ -59,7 +60,7 @@ func Upsert[T any](ctx context.Context, engine *Engine, object T,
 	engine.mutex.Lock()
 	defer engine.mutex.Unlock()
 	if len(nsId) > 1 {
-		return 0, object, false, fmt.Errorf("only one namespace is allowed")
+		return 0, object, false, errors.New("only one namespace is allowed")
 	}
 
 	ctx, ns, err := getDefaultNamespace(ctx, engine, nsId...)
@@ -131,7 +132,7 @@ func Get[T any, R UniqueField](ctx context.Context, engine *Engine, uniqueField 
 	defer engine.mutex.Unlock()
 	var obj T
 	if len(nsId) > 1 {
-		return 0, obj, fmt.Errorf("only one namespace is allowed")
+		return 0, obj, errors.New("only one namespace is allowed")
 	}
 	ctx, ns, err := getDefaultNamespace(ctx, engine, nsId...)
 	if err != nil {
@@ -142,10 +143,20 @@ func Get[T any, R UniqueField](ctx context.Context, engine *Engine, uniqueField 
 	}
 
 	if cf, ok := any(uniqueField).(ConstrainedField); ok {
-		return getByConstrainedField[T](ctx, ns, cf)
+		objType := reflect.TypeOf(obj)
+		sch, err := getSchema(ctx, ns)
+		if err != nil {
+			return 0, obj, err
+		}
+		for _, t := range sch.Types {
+			if t.Name == objType.Name() {
+				return getByConstrainedField[T](ctx, ns, cf)
+			}
+		}
+		return 0, obj, errors.New("type not found")
 	}
 
-	return 0, obj, fmt.Errorf("invalid unique field type")
+	return 0, obj, errors.New("invalid unique field type")
 }
 
 func Query[T any](ctx context.Context, engine *Engine, queryParams QueryParams,
@@ -153,7 +164,7 @@ func Query[T any](ctx context.Context, engine *Engine, queryParams QueryParams,
 	engine.mutex.Lock()
 	defer engine.mutex.Unlock()
 	if len(nsId) > 1 {
-		return nil, nil, fmt.Errorf("only one namespace is allowed")
+		return nil, nil, errors.New("only one namespace is allowed")
 	}
 	ctx, ns, err := getDefaultNamespace(ctx, engine, nsId...)
 	if err != nil {
@@ -169,7 +180,7 @@ func Delete[T any, R UniqueField](ctx context.Context, engine *Engine, uniqueFie
 	defer engine.mutex.Unlock()
 	var zeroObj T
 	if len(nsId) > 1 {
-		return 0, zeroObj, fmt.Errorf("only one namespace is allowed")
+		return 0, zeroObj, errors.New("only one namespace is allowed")
 	}
 	ctx, ns, err := getDefaultNamespace(ctx, engine, nsId...)
 	if err != nil {
@@ -207,5 +218,5 @@ func Delete[T any, R UniqueField](ctx context.Context, engine *Engine, uniqueFie
 		return uid, obj, nil
 	}
 
-	return 0, zeroObj, fmt.Errorf("invalid unique field type")
+	return 0, zeroObj, errors.New("invalid unique field type")
 }

--- a/unit_test/api_test.go
+++ b/unit_test/api_test.go
@@ -59,6 +59,14 @@ func TestFirstTimeUser(t *testing.T) {
 	require.Equal(t, "A", queriedUser2.Name)
 	require.Equal(t, "123", queriedUser2.ClerkId)
 
+	// Search for a non-existent record
+	_, _, err = modusdb.Get[User](context.Background(), engine, modusdb.ConstrainedField{
+		Key:   "clerk_id",
+		Value: "456",
+	})
+	require.Error(t, err)
+	require.Equal(t, "no object found", err.Error())
+
 	_, _, err = modusdb.Delete[User](context.Background(), engine, gid)
 	require.NoError(t, err)
 
@@ -66,7 +74,25 @@ func TestFirstTimeUser(t *testing.T) {
 	require.Error(t, err)
 	require.Equal(t, "no object found", err.Error())
 	require.Equal(t, queriedUser3, User{})
+}
 
+func TestGetBeforeObjectWrite(t *testing.T) {
+	ctx := context.Background()
+	engine, err := modusdb.NewEngine(modusdb.NewDefaultConfig(t.TempDir()))
+	require.NoError(t, err)
+	defer engine.Close()
+	ns, err := engine.CreateNamespace()
+	require.NoError(t, err)
+
+	_, _, err = modusdb.Get[User](ctx, engine, uint64(1), ns.ID())
+	require.Error(t, err)
+
+	_, _, err = modusdb.Get[User](ctx, engine, modusdb.ConstrainedField{
+		Key:   "name",
+		Value: "test",
+	}, ns.ID())
+	require.Error(t, err)
+	require.Equal(t, "type not found", err.Error())
 }
 
 func TestCreateApi(t *testing.T) {
@@ -125,6 +151,14 @@ func TestCreateApiWithNonStruct(t *testing.T) {
 	_, _, err = modusdb.Create[*User](context.Background(), engine, &user, ns1.ID())
 	require.Error(t, err)
 	require.Equal(t, "expected struct, got ptr", err.Error())
+
+	_, _, err = modusdb.Create[[]string](context.Background(), engine, []string{"foo", "bar"}, ns1.ID())
+	require.Error(t, err)
+	require.Equal(t, "expected struct, got slice", err.Error())
+
+	_, _, err = modusdb.Create[float32](context.Background(), engine, 3.1415, ns1.ID())
+	require.Error(t, err)
+	require.Equal(t, "expected struct, got float32", err.Error())
 }
 
 func TestGetApi(t *testing.T) {

